.github/actions/run_local_pytest/action.yml

```diff
@@ -0,0 +1,48 @@
+name: 'Install Local Python dependencies'
+description: 'Install Local Python dependencies'
+inputs:
+  job_type:          {default: '', type: string, description: Specifies which steps should be executed}
+  build_type:        {default: 'release', type: string, description: Specifies whether the build that we are testing is release or debug}
+  threads:           {default: '1', type: string, description: Specifies the number of threads that will be used for tests }
+  fast_tests_only:   {default: '0', type: string, description: Specifies whether we want to run only the fast tests or not }
+  other_params:      {default: '', type: string, description: Other params that should be passed to the python command }
+runs:
+  using: "composite"
+  steps:
+    - name: Install local dependencies with pip
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip
+
+        python >build_tooling/requirements-analysis-workflow.txt <<END
+        from configparser import ConfigParser
+        cf = ConfigParser()
+        cf.read("setup.cfg");
+
+        # setup_requires is not needed
+        print(cf['options']['setup_requires'])
+        print(cf['options']['install_requires'])
+        print(cf['options.extras_require']['Testing'])
+        END
+
+        python -m pip install -r build_tooling/requirements-analysis-workflow.txt
+        python setup.py protoc --build-lib python
+
+    - name: Run local pytests
+      if: inputs.job_type == 'run'
+      shell: bash
+      run: |
+        yum install nodejs npm -y
+        npm config set strict-ssl false
+        npm install -g azurite
```

Comment:
Above three lines should be baked into the container (if not already).

Reply:
i don't think that this is in the scope of this task, I have opened a new issue to track this [here](https://github.com/man-group/ArcticDB/issues/1036)

Reply:
I am not sure how Alma Linux would help? It certainly wouldn't have `azurite` out of the box.

You can just move these three lines [here](https://github.com/man-group/ArcticDB/blob/master/build_tooling/build_many_linux_image.sh#L47)

Reply:
As I have mentioned in the ticket, these dependencies should be added to the container, we don't expect them to be there out-of-the-box.
I don't want to do it in this PR, as it would have impact that is outside of the scope of this task and this PR has already become far too bloated.